operator/watchers/service_sync.go

```diff
@@ -152,12 +199,17 @@ func (s *serviceSync) loop(ctx context.Context, health cell.Health) error {
 				}
 			case resource.Upsert:
 				svc := ev.Object
-				cs, toUpsert := converter.Convert(svc, getEndpoints)
-				if toUpsert {
-					upsert(cs)
-				} else {
-					s.store.DeleteKey(ctx, cs)
+				// Check namespace global status before syncing
+				if !isNamespaceGlobal(svc.Namespace) {
+					s.Log.Debug("Skipping service sync for non-global namespace",
+						logfields.K8sSvcName, svc.Name,
+						logfields.K8sNamespace, svc.Namespace,
+					)
+					// Delete the service from kvstore if it exists (namespace may have become non-global)
+					s.store.DeleteKey(ctx, converter.ForDeletion(svc))
```

Comment:
Is this required for service? The namespace field is immutable, so it cannot change unless deleted (which will delete from `kvstore`). If namespace becomes local, the namespace event will handle this deletion.